packages/gateway/src/features/user/usersAPI.ts

```diff
@@ -61,11 +65,15 @@ export class UsersAPI extends RESTDataSource {
       return cachedResponse
     }
 
-    const response = this.post('getUser', { mobile })
+    try {
+      const response = this.post('getUser', { mobile })
```

Comment:
would this try-catch also eat up the error if this.post fails for other reason than 404? think we should handle this more granularly

Reply:
I think it's fine to leave the call here, but agree on the memoization, it might cause issues if the user data changes and the cache isn't updated. Likely the cache doesn't help performance either, as far as I can see "fetchUserDetailsByEmail" is only called from Settings view.